smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/KafkaSourceWithLegacyMetadataTest.java

```diff
@@ -0,0 +1,101 @@
+package io.smallrye.reactive.messaging.kafka;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import io.smallrye.reactive.messaging.health.HealthReport;
+import io.smallrye.reactive.messaging.kafka.api.KafkaMetadataUtil;
+import io.smallrye.reactive.messaging.kafka.base.KafkaMapBasedConfig;
+import io.smallrye.reactive.messaging.kafka.base.KafkaTestBase;
+import io.smallrye.reactive.messaging.kafka.impl.KafkaSource;
+
+/**
+ * Duplicate of {@link KafkaSourceTest} - delete once we remove the legacy
+ * {@link io.smallrye.reactive.messaging.kafka.api.IncomingKafkaRecordMetadata}
+ * <p/>
+ * Tests not using metadata have been removed
```

Comment:
The main reason for this is that `testABeanConsumingTheKafkaMessagesStartingOnFifthOffsetFromLatestThatFailsUntilSecondRebalance()`  ended up with 
Gets ` [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]` rather than ` [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]`.

It worked fine on its own, so again I think it is due to some pollution in the test setup. But I removed all non-metadata tests rather than investigate.

Reply:
Did the topic get reused no?